openmdao/core/tests/test_distrib_derivs.py

```diff
@@ -307,8 +309,326 @@ def test_fan_in_grouped(self, nlsolver):
         assert_near_equal(J['C4.y', 'P1.x'], diag1, 1e-6)
         assert_near_equal(J['C4.y', 'P2.x'], diag2, 1e-6)
 
-    def test_distrib_voi(self):
-        raise unittest.SkipTest("distrib vois no supported yet")
+    def test_distrib_voi_dense(self):
+        size = 7
+
+        prob = om.Problem()
+        model = prob.model
+
+        ivc = om.IndepVarComp()
+        ivc.add_output('x', np.ones((size, )))
+        ivc.add_output('y', np.ones((size, )))
+        ivc.add_output('a', -3.0 + 0.6 * np.arange(size))
+
+        model.add_subsystem('p', ivc, promotes=['*'])
+        model.add_subsystem("parab", DistParab(arr_size=size, deriv_type='dense'), promotes=['*'])
+        model.add_subsystem('sum', om.ExecComp('f_sum = sum(f_xy)',
+                                               f_sum=np.ones((size, )),
+                                               f_xy=np.ones((size, ))),
+                            promotes=['*'])
+
+        model.add_design_var('x', lower=-50.0, upper=50.0)
+        model.add_design_var('y', lower=-50.0, upper=50.0)
+        model.add_constraint('f_xy', lower=0.0)
```

Comment:
I'm a little bit nervous about only having one distributed constraint tested in all cases. Might be good to have an alternate formulation which tracks more than one dist constraint (catches potential nonzero index edge cases)

Reply:
That is a good point. I'll put together something with a larger set of constraints.

Reply:
Got a new test added with 2 distributed constraints and 2 regular ones.